docs/adrs/0009-nimbus-web-sdk-architecture.md

```diff
@@ -0,0 +1,139 @@
+# Nimbus Web SDK Architecture
+
+* Status: proposed
+* Deciders: James Hugman, Charlie Humphreys, Yashika Khurana, Jared Lockhart
+* Date: 2023-15-03
+
+Technical Story: https://mozilla-hub.atlassian.net/browse/EXP-3158
+
+## Context and Problem Statement
+
+This is part of a body of work necessary to support the use of Nimbus within web applications. 
+The current Nimbus SDK is written in such a way that it supports client-oriented experimentation — experiments are downloaded, evaluated, and stored on the client, and a feature store is exposed with the experiment branches applied. 
+In order to support web experimentation, the Nimbus SDK will need to be updated to be stateless, to support a more statically defined set of helper methods, and to have additional support for Python.
+
+Ultimately, the problem we're trying to solve can be boiled down to one question — how can we update the Nimbus SDK to support web applications while continuing to support the existing clients?
+
+As an example of what the Cirrus API might look like, we can likely expect endpoints to perform the following:
+* Enroll a user into available experiment(s)
+  * This would return the enrolled experiments as well as the feature values given the enrollments
+* Fetch the default feature values
+* Fetch the feature manifest
+* Fetch a specific feature value given enrolled experiments
+
+## Decision Drivers
+
+* Architecture decisions for Nimbus on the web[<sup>[1.i]</sup>](#links)
+  * The core logic of the existing SDK (Rust) will be re-used for the Nimbus web service (Cirrus)
+  * The SDK needs to support Python via UniFFI
+  * The SDK needs to be stateless
+
+## Considered Options
+
+* Separate `nimbus-core`, `nimbus-sdk`, and `cirrus-sdk` libraries
+* Cargo features, one library
+* Cargo features, separate library
+* Expose a stateless API alongside the stateful API (Do nothing)
+
+## Decision Outcome
+
+We have decided to move forward with option number 2, "Cargo features, one library".
+This option, like the others mentioned, meets the key decision drivers.
+We believe using this option will be the most maintainable long-term, despite the added complexity of using cargo features.
+In addition, implementing this option has a similarly short timeline and amount of work necessary as compared to the "Cargo features, separate library" option, without the additional overhead of more complex typing.
+
+### Positive Consequences
+
+* Meets all decision drivers.
+* Small amount of work necessary with a very limited amount of code churn.
+
+### Negative Consequences
+
+* It will be difficult to draw boundaries between Cirrus and Nimbus code.
+* We could run into unexpected issues with UniFFI, as multiple UDL files in a single crate has not yet been tested. 
```

Comment:
Hm that's a really good point. I'm certainly not opposed, it would work incredibly well in tandem with the feature attributes.

I think it's a great plan but I also don't know the time investment that would require. I'm going to start by setting up the cargo features first with the proc macro piece being an optional thing as we move forward with the project and add the cirrus features.